cmd/object-api-common.go

```diff
@@ -289,6 +289,7 @@ func listObjects(ctx context.Context, obj ObjectLayer, bucket, prefix, marker, d
 		if !ok {
 			// Closed channel.
 			eof = true
+			break
```

Comment:
I have no idea if this doesn't break anything along the way. If the unit tests are successful this is probably okay.

Reply:
This is probably fine. I'll wait for the test results.